src/main/java/org/springframework/data/elasticsearch/core/mapping/SimpleElasticsearchPersistentEntity.java

```diff
@@ -73,6 +74,7 @@
 	private @Nullable ElasticsearchPersistentProperty joinFieldProperty;
 	private @Nullable VersionType versionType;
 	private boolean createIndexAndMapping;
+	private Dynamic dynamic = Dynamic.INHERIT;
```

Comment:
```suggestion
	private final Dynamic dynamic = Dynamic;
```

I'd prefer a `final`property